core/shared/platform/windows/win_time.c

```diff
@@ -18,3 +52,127 @@ os_time_get_boot_microsecond()
 
     return ((uint64)ts.tv_sec) * 1000 * 1000 + ((uint64)ts.tv_nsec) / 1000;
 }
+
+uint64
+os_clock_res_get(bh_clock_id_t clock_id, uint64 *resolution)
+{
+    switch (clock_id) {
+        case BH_CLOCK_MONOTONIC:
+        {
+            const uint64 result =
+                (uint64)NANOSECONDS_PER_SECOND
+                          / calculate_monotonic_clock_frequency();
+            *resolution = result;
+            return BHT_OK;
+        }
+        case BH_CLOCK_REALTIME:
+        case BH_CLOCK_PROCESS_CPUTIME_ID:
+        case BH_CLOCK_THREAD_CPUTIME_ID:
+        {
+            PULONG MaximumTime;
+            PULONG MinimumTime;
+            PULONG CurrentTime;
+            NTSTATUS
+            NTAPI Res;
+            if (Res = NtQueryTimerResolution(&MaximumTime, &MinimumTime,
+                                             &CurrentTime)) {
+                !NT_SUCCESS(Res);
+            }
+            else
+            
+            {
+                return NT_ERROR(Res);
+            }
+        }
+    }
+    
+
+}
+
+uint64
+os_clock_time_get(bh_clock_id_t clock_id, uint64 precision,
+                     uint64 *time)
+{
+    switch (clock_id) {
+
+        case BH_CLOCK_REALTIME:
+        {
+            FILETIME SysNow;
+#if NTDDI_VERSION >= NTDDI_WIN8
+            GetSystemTimePreciseAsFileTime(&SysNow);
+#else
+            GetSystemTimeAsFileTime(&SysNow);
+
+#endif
+            time = from_file_time_to_wasi_timestamp(SysNow);
+            return 0;
+        }
+
+        case BH_CLOCK_MONOTONIC:
+        {
+            uint64_t Nanoseconds;
+            const auto Counter = calculate_monotonic_clock_frequency();
+            if (NANOSECONDS_PER_SECOND
+                           % calculate_monotonic_clock_frequency()
+                       == 0) {
+                Nanoseconds = Counter
```

Comment:
Is this calculation correct? Looks like we only use clock frequency to calculate the time?